src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,13 +31,102 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        setValues(address);
+    }
+    
+    /**
+     * Represents a Person's block number in the address.
+     */
+    class Block{
+    	private final int blockNo_; 
+    	
+    	Block (int blockNo){
+    		this.blockNo_ = blockNo; 
+    	}
+    	
+    	/**
+    	 * Getter method for blockNo_.
+    	 * @returns int - block number of address 
+    	 */
+    	public int getBlockNo_ (){
+    		return blockNo_;
+    	}
+    }
+    
+    /**
+     * Represents a Person's street name in the address.
+     *
+     */
+    class Street{
+    	private final String streetName_; 
+    	
+    	Street (String streetName){
+    		this.streetName_ = streetName;
+    	}
+    	
+    	/**
+    	 * Getter method for streetName_.
+    	 * @returns String - street name of address
+    	 */
+    	public String getStreetName_(){
+    		return streetName_;
+    	}
+    }
+    
+    /**
+     * Represents a Person's unit number in the address.
+     */
+    class Unit{
+    	private final int unitNo_;
+    	
+    	Unit (int unitNo){
+    		this.unitNo_ = unitNo;
+    	}
+    	
+    	/**
+    	 * Getter method for unitNo_.
+    	 * @returns int - unit number of address
+    	 */
+    	public int getUnitNo_(){
+    		return unitNo_;
+    	}
+    	
+    }
+    
+    
+    /**
+     * Represents a Person's postal code in the address.
+     *
+     */
+    class PostalCode{
+    	private final int postalCode_;
+    	
+    	PostalCode (int postalCode){
+    		this.postalCode_ = postalCode;
+    	}
+    	
+    	/**
+    	 * Getter method for postalCode_.
+    	 * @returns int - postal code of address
+    	 */
+    	public int getPostalCode_(){
+    		return postalCode_;
+    	}
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        return test.matches(ADDRESS_VALIDATION_REGEX) && test.split(",").length==4;
+    }
```

Comment:
Why is the length check for the value 4? its not clear to someone seeing your code for the first time to understand the significance of the number 4. Please store constants in meaning variable names.